backend/api/src/add-subsidy.ts

```diff
@@ -51,33 +53,20 @@ export const addContractLiquidity = async (
       fromType: 'USER',
     })
 
-    const liquidity = await firestore.runTransaction(async (transaction) => {
-      const contractDoc = firestore.doc(`contracts/${contractId}`)
+    const subsidyAmount = (1 - SUBSIDY_FEE) * amount
 
-      const newLiquidityProvisionDoc = firestore
-        .collection(`contracts/${contractId}/liquidity`)
-        .doc()
+    const { newLiquidityProvision, newTotalLiquidity, newSubsidyPool } =
+      getNewLiquidityProvision(userId, subsidyAmount, contract)
```

Comment:
I think `newSubsidyPool` can be changed to `subsidyPoolIncrease` or something, which then can use firestore's increment() to update the contract subsidyPool field that would not cause a problem with multiple subsidies being added at the same time.

Reply:
Same for `newTotalLiquidity`